samza-core/src/main/java/org/apache/samza/startpoint/StartpointKey.java

```diff
@@ -41,12 +41,29 @@
     this.taskName = taskName;
   }
 
+  @JsonIgnore
```

Comment:
The idea of including only relevant fields from `StartpointKey` to reduce redundancy is a good one. 

However, rather than marking individual fields in model POJO as included or not, it would be better to have a custom `JsonSerializer` implementation and have this logic to include fields we want in it. Similar to [SystemStreamPartitionSerializer](https://github.com/apache/samza/blob/master/samza-core/src/main/java/org/apache/samza/serializers/model/SamzaObjectMapper.java#L203)

What do you think?

Reply:
@shanthoosh good idea. I'm always for being explicit. Please see the latest commit.